collection/lens.rkt

```diff
@@ -39,6 +67,21 @@
   (make-lens (λ (target) (nth target index))
              (λ (target value) (set-nth target index value))))
 
+(define (mapper-lens lens)
```

Comment:
Indeed, and that is precisely why I think it *should* be called either `map-lens` or `lens-map`.

Traditionally, `map`'s type is `(a -> b) -> f a -> f b`. The type of `map-lens` is `(Lens l) => l a b -> f a -> l (f a) (f b)` (approximately, I'm fudging the types a little bit here).

Fundamentally, `map-lens` is the analogue of `map` for lenses in every way. Traditionally, `map` is a one-way operation, so generalizing it to be two-way means, of course, producing a lens rather than a procedure when partially applied.